Commands/SitesMassWordPressUpdateCommand.php

```diff
@@ -20,850 +21,1021 @@
+		//	Commit all changes
+		if ( ! $report && $commit ) {
 
-      return false;
+			$this->execute(
+				'echo y | terminus site code commit --site='. $name .' --env='. $environ .' --message="Updates applied by Mass Wordpress Update terminus plugin."',
+				true,
+				true
+			);
 
-    }
+		}
 
-    require_once 'lib/Spyc.php';
 
-    $yaml_settings = spyc_load_file( $dir );
+		//	Doing auto-deploy
+		if ( $proceed && ( $this->isAutoDeploy() || $this->yamlGetSiteSetting( $name, 'auto-deploy' ) ) ) {
 
-    $yaml_settings = $this->validateYamlSettings( $yaml_settings );
+			// Set connection back to sftp.
+			if ( $mode == 'git' ) {
 
-    return $yaml_settings;
+				$mode = $this->setSiteConnectionMode( $name, $environ, 'sftp' );
 
-  }
+			}
 
-  private function yamlGetSiteArgs( $name, $yaml ){
+			$autoDeployTest = $this->auto_deploy( $name, 'test' );
 
-    if ( ! isset( $yaml['sites']['update'] ) ) return array();
+			$autoDeployLive = $this->auto_deploy( $name, 'live' );
 
-    $args = false;
+		}
 
-    foreach ($yaml['sites']['update'] as $key => $value) {
 
-      if ( $value['name'] === $name ) {
+		//	Completion message.
+		if ( ! $report ) {
 
-        $args = $value;
+			$this->log()->notice( 'Finished plugins updates for {environ} environment of {name} site.', array(
+				'environ' => $environ,
+				'name' => $name,
+			) );
 
-        if ( isset( $yaml['sites']['settings'] ) ) {
+		}
 
-          $args = array_merge( $yaml['sites']['settings'], $args );
 
-        }
+		//	Send slack message
+		if ( isset( $this->yaml_settings['slack_settings'] ) ) {
 
-      }
+			$date = new \DateTime();
 
-    }
+			$slack_notif_text = "";
+			$slack_notif_text .= "*Terminus update on " . $name . "*\n\n";
 
-    return $args;
+			if ( $update_report ) {
 
-  }
+				$slack_notif_text .= "```";
 
-  private function yamlGetSiteSetting( $name, $setting ){
+				foreach ( $update_report as $report ) {
 
-    if ( ! $this->yaml_settings ) return false;
+					$slack_notif_text .= $report . "\n";
 
-    $settings = $this->yamlGetSiteArgs( $name, $this->yaml_settings );
+				}
 
-    foreach ($settings as $key => $value) {
+				$slack_notif_text .= "```";
 
-      if ( $key === $setting ) {
+			}
 
-        return $value;
+			$slack = simple_slack( $this->slack_settings['url'], array(
+				'username'		=> $this->slack_settings['username'],
+				'channel'		=> $this->slack_settings['channel'],
+				'icon_emoji'	=> $this->slack_settings['icon_emoji'],
+				'text'			=> $slack_notif_text,
+				'mrkdwn'		=> true,
+				'attachments'	=> array( array(
+					'fallback'		=> 'Terminus MWU message.',
+					'color'			=> '#ffb305',
+					'author_name'	=> 'Terminus MWU',
+					'fields'		=> array(
+						array(
+							'title'		=> 'Dashboard',
+							'value'		=> 'https://dashboard.pantheon.io/sites/',
+							'short'		=> true
+						),
+						array(
+							'title'		=> 'Dev',
+							'value'		=> ( isset( $env_url['dev'] ) ? $env_url['dev'] : 'no link provided.' ),
```

Comment:
These values from here can easily be guest since they are all the same: `http://dev-[site-name].pantheonsite.io/`, so there is no need to put those values in the yml. What we can do is to include an `url` one, for live. Since sometimes you have multiples domains on a live website.